typing/types.ml

```diff
@@ -79,6 +79,38 @@ module TypeOps = struct
   let equal t1 t2 = t1 == t2
 end
 
+(* *)
+
+module Uid = struct
+  type t =
+    | Compilation_unit of string
+    | Item of { comp_unit: string; id: int }
+    | Internal
+    | Predef of string
+
+  let mk =
+    let id = ref (-1) in
```

Comment:
- I think the generator should be reset (or otherwise made local to the compilation_unit), so that compiling multiple units in a single invocation of the compiler produces the same .cmi files.

- It would be better to provide custom equality and hash functions rather than relying on the generic operations(through Hashtbl in Env).

Reply:
Ack for `reset`, I'll push something for that.

AS for custom operations: I'm fine with providing a custom equality function, but for hashing, I can't see myself doing anything better the generic `hash` function.